### DISCUSSION TITLE
Exporting table data from custom SQL query

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11835] [Help] [answered]
Hi,

We have the following scenario:

A custom MSSQL query, connected to a BB table.

A user needs to be able to export all data from that table to a CSV file, at the same time I want BB to update a status column for each row exported.

We can achieve this if we connect the data provider directly to an SQL table, but when connecting to a custom query the row selection option on the table does not bring up the checkboxes to select a row.

Is there currently anyway to achieve this within Budibase?

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @benaunt 

Exporting table rows only works for data tables, such as fetched SQL tables, but will not work for custom SQL queries. 

If you need to update the status of each column exported, you could try the following flow:

1. Trigger an automation that updates the status of each selected row (there's a [great example here](https://github.com/Budibase/budibase/discussions/11055) on how to do that)

2. Navigate to a screen specifically for exporting the rows, which will now have been updated. You will need to use [Sync automations](https://docs.budibase.com/docs/synchronous-automations) to wait for the row updates to complete. It's also worth noting that bulk row update is fairly slow at the moment, so this would only be suitable for modest numbers of selected rows.

3. Add a standard export rows button within this new screen. At this point you know the statuses have been updated.
